src/common/error/src/ext.rs

```diff
@@ -88,6 +88,52 @@ impl crate::snafu::ErrorCompat for BoxedError {
     }
 }
 
+#[derive(Debug)]
+pub struct NeitherError {
```

Comment:
is this Either or Neither?

Reply:
I struggled on this too 🤣 This struct is to express that both two errors occur, thus non branch success -> neither. Maybe there is a better name?

Reply:
How about `CompoundError`? `compound` means it has two or more children. iirc, `neither` means neither `one` nor `another`.

Reply:
Sounds great 👍